### PR TITLE
Add api.py

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: lmdeploy/api.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+
+
+def pipeline(model_path, instance_num=32, tp=1, log_level='ERROR', **kwargs):
+    """
+    Args:
+        model_path (str): the path of the deployed model
```
Comment (lvhan028) [inline lmdeploy/api.py:8]: may provide an example

**Comment by lvhan028** [comment]
pipeline支持流式吗？

**Comment by AllentDan** [comment]
> pipeline支持流式吗？

没

File: lmdeploy/api.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        server_name (str): host ip for serving
+        server_port (int): server port
+        instance_num (int): number of instances of turbomind model
+        tp (int): tensor parallel
+        log_level(str): set log level whose value among [CRITICAL, ERROR, WARNING, INFO, DEBUG]
+
+    Return:
+        APIClient: A client chatbot for LLaMA series models.
+
+    Examples:
```
Comment (irexyc) [inline lmdeploy/api.py:85]: 每创建一次client都会启动一个serve么？

**Review by irexyc — APPROVED** [review]

File: lmdeploy/api.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+from typing import Optional
+
+
+def pipeline(model_path,
```
Comment (RunningLeon) [inline lmdeploy/api.py:6]: may add the type hint to all arguments.

File: lmdeploy/serve/async_engine.py [hunk 4]
```diff
@@ -20,14 +20,35 @@ class AsyncEngine:
...
+        model_name (str): needed when model_path is a pytorch model on
+            huggingface.co, such as "InternLM/internlm-chat-7b",
+            "Qwen/Qwen-7B-Chat ", "baichuan-inc/Baichuan2-7B-Chat" and so on.
         instance_num (int): instance numbers to be created
         tp (int): tensor parallel
     """
 
-    def __init__(self, model_path, instance_num=32, tp=1, **kwargs) -> None:
+    def __init__(self,
+                 model_path,
```
Comment (RunningLeon) [inline lmdeploy/serve/async_engine.py:44]: may add the type hint to all args

File: lmdeploy/__init__.py [hunk 5]
```diff
@@ -1,4 +1,10 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+
+from lmdeploy.api import client, pipeline, serve
+
+__all__ = ['pipeline', 'serve', 'client']
```
Comment (lvhan028) [inline lmdeploy/__init__.py:5]: If we put `serve` here, should we add the serve dependencies to runtime.txt?

Comment (AllentDan) [reply lmdeploy/__init__.py:5]: No idea here. @RunningLeon Any comments?

Comment (RunningLeon) [reply lmdeploy/__init__.py:5]: `pipeline`, `serve`, `client` all involve codes from `lmdeploy.serve`. Maybe we add a note in docs to let users install lmdeploy by `pip install lmdeploy[serve]` when using this api?

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**